### PR TITLE
Feature.epoll

### PR DESCRIPTION
**Pull request by yakman2020**
Implement epoll on linux and needed stubs on windows.

### CODE REVIEW

**Comment by yakman2020** [comment]
bors try

**Comment by yakman2020** [comment]
bors try

**Comment by johnkord** [comment]
This resolves part of #2047

**Comment by johnkord** [comment]
@radhikaj Do you think you could schedule time for someone to review this PR? Thank you so much!

File: include/openenclave/bits/module.h [hunk 1]
```diff
@@ -63,6 +63,17 @@ oe_result_t oe_load_module_host_socket_interface(void);
  */
 oe_result_t oe_load_module_host_resolver(void);
 
+/**
+ * Load the event polling module epoll module.
```
Comment (mikbras) [inline include/openenclave/bits/module.h]: Typo: minor typo in public API.

File: include/openenclave/internal/syscall/sys/bits/epoll_event.h [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#if !defined(__aarch64__)
```
Comment (mikbras) [inline include/openenclave/internal/syscall/sys/bits/epoll_event.h]: Maybe a brief comment about how this packing it to enforce consistency with the related struct in MUSL.

File: libc/epoll.c [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/internal/syscall/sys/epoll.h>
+#include <sys/epoll.h>
+
+OE_STATIC_ASSERT(sizeof(struct oe_epoll_event) == sizeof(struct epoll_event));
+// OE_CHECK_FIELD(struct oe_epoll_event, struct epoll_event, data);
```
Comment (mikbras) [inline libc/epoll.c]: Maybe uncomment this and see if it fails. This constraint should pass.

File: include/openenclave/bits/module.h (hunk 1 above)
Comment (yakman2020) [reply include/openenclave/bits/module.h]: done

File: include/openenclave/internal/syscall/sys/bits/epoll_event.h (hunk 2 above)
Comment (yakman2020) [reply include/openenclave/internal/syscall/sys/bits/epoll_event.h]: done

File: libc/epoll.c (hunk 3 above)
Comment (yakman2020) [reply libc/epoll.c]: Needed to readd the OE_CHECK_FEILD macro. Done.

**Comment by johnkord** [comment]
Thanks so much everyone for your reviews!

bors r+

**Comment by johnkord** [comment]
Okay, CLA bot looks good now. Let's try that again :)

bors r+

**Pull request merged**